feat: add new ACs from the AC review meeting (#1754)
vegaprotocol/specs: protocol/0081-SUCM-successor_markets.md

A new market proposal sets parent market Id to a market that has settled. The parent market has non-zero insurance pool balance. If the new market clears the opening auction before `parent settlement time + market.value.windowLength` then the virtual stakes are carried over and the relevant fraction of the insurance pool is transferred over (<a name="0081-SUCM-006" href="#0081-SUCM-006">0081-SUCM-006</a>).

A new market proposal sets parent market Id to a market that has settled. The parent market has non-zero insurance pool balance. If the new market clears the opening auction after `parent settlement time + market.value.windowLength` then no virtual stakes are carried over, there is no transfer into the insurance pool of the new market from the parent and the new market has no parent market Id set (<a name="0081-SUCM-007" href="#0081-SUCM-007">0081-SUCM-007</a>)

> Successor markets can't be enacted if the parent market is still in the "proposed" state (<a name="0081-SUCM-008" href="#0081-SUCM-008">0081-SUCM-008</a>)

Reviewer: I would like this to be made very clear @davidsiska-vega, because IIRC, there shouldn't be anything to prevent a successor market from being enacted/leaving opening auction here, it's just that it would be a rather pointless thing to do, because there simply would be no ELS to carry over from the parent market.

The question here would then be: should the successor market be rejected, or should we clear the parent ID (ie enact this successor as a separate market)?

Reviewer 2: This AC isn't correct and should be removed from the PR.

Reviewer 2: The other option is to modify the AC to say that 
```
Successor markets can be enacted if the parent market is still in the "proposed" state. There is no virtual stake to copy over and no insurance pool balance to transfer. 
```

> Successor markets can't be enacted if the parent market is still in the "proposed" state (<a name="0081-SUCM-008" href="#0081-SUCM-008">0081-SUCM-008</a>)

Author: ```suggestion
Successor markets can be enacted if the parent market is still in the "proposed" state. There is no virtual stake to copy over and no insurance pool balance to transfer  (<a name="0081-SUCM-008" href="#0081-SUCM-008">0081-SUCM-008</a>)
```

Successor markets can be enacted when the parent market is in opening auction (<a name="0081-SUCM-009" href="#0081-SUCM-009">0081-SUCM-009</a>)

> A successor market proposal can be enacted when the parent market is in one of the following states: Suspended, Active, Trading terminated or Settled (settled within the successor time window) (<a name="0081-SUCM-010" href="#0081-SUCM-010">0081-SUCM-010</a>)

Reviewer 2: Yes, but it doesn't mean that a successor market proposal cannot be enacted when the parent market is in other states. E.g. 
- parent Settled and past the successor time window... fine but pointless no virtual stake to copy, no insurance pool balance to transfer. 
- proposed, see above.

Reviewer: IIRC, there's some ACs covering the settled+ past successor window elsewhere. I think the different states mentioned here are more about explicitly stating that successors can get enacted when parent is in one of these states, without issues

Reviewer 3: - I thought we had one for when the time window has already expired... We definitely should.
- - 0081-SUCM-007 covers what happens if it expires during the child proposal
- - We probably need another one to cover it already having expired prior to the child proposal: "Propose a successor market which specifies a parent which is settled, and for which the successor time window has expired". Expect pass/fail?
- I think Proposed is covered in 0081-SUCM-008

Reviewer 2: @jgsbennett I've added a suggestion below to cover the middle bullet point. https://github.com/vegaprotocol/specs/pull/1754/files#r1226491529

> When a successor market is enacted, all other proposals (pending or proposed) are automatically rejected. Any LP submissions associated with these proposals are cancelled, and the funds are released (<a name="0081-SUCM-011" href="#0081-SUCM-011">0081-SUCM-011</a>)

Reviewer 2: Can we please rephrase `all other proposals (pending or proposed) are automatically rejected`. It's simply not correct. There can be many proposals that have nothing to do with the successor or the market and those should be left alone. 

Proposals for a market that name the same parent should be rejected.

> When a successor market is enacted, all other proposals (pending or proposed) are automatically rejected. Any LP submissions associated with these proposals are cancelled, and the funds are released (<a name="0081-SUCM-011" href="#0081-SUCM-011">0081-SUCM-011</a>)

Reviewer 2: ```suggestion
When a successor market is enacted (i.e. leaves the opening auction), all other proposals (pending or proposed) are automatically rejected. Any LP submissions associated with these proposals are cancelled, and the funds are released (<a name="0081-SUCM-011" href="#0081-SUCM-011">0081-SUCM-011</a>)
```

> When a successor market is enacted (i.e. leaves the opening auction), all other proposals (pending or proposed) are automatically rejected. Any LP submissions associated with these proposals are cancelled, and the funds are released (<a name="0081-SUCM-011" href="#0081-SUCM-011">0081-SUCM-011</a>)

Author: ```suggestion
When a successor market is enacted (i.e. leaves the opening auction), all other related successor market proposals, in the state "pending" or "proposed", are automatically rejected. Any LP submissions associated with these proposals are cancelled, and the funds are released (<a name="0081-SUCM-011" href="#0081-SUCM-011">0081-SUCM-011</a>)
```

Author: @davidsiska-vega - does this rewording make it more clear / correct?

> With two successor markets in opening auction, that have the same parent market, and one additional market in the state "Proposed". Enact one of the two markets in opening auction. The other market in auction and the proposed market should both be "Rejected" and all LP funds will be released (<a name="0081-SUCM-014" href="#0081-SUCM-014">0081-SUCM-014</a>)

Reviewer 2: ```suggestion
With two successor markets in opening auction, that have the same parent market, and one additional market in the state "Proposed". Get one of the two markets to leave the opening auction (passage of time, LP commitment, crossing trade). The other market in auction and the proposed market should both be "Rejected" and all LP funds will be released (<a name="0081-SUCM-014" href="#0081-SUCM-014">0081-SUCM-014</a>)
```

### APIs

> It is possible to fetch a market "family tree" containing the initial market and the full successor line (<a name="0081-SUCM-012" href="#0081-SUCM-012">0081-SUCM-012</a>)

Reviewer 2: ```suggestion
It is possible to fetch a market "parent / successor chain" containing the initial market and the full successor line (<a name="0081-SUCM-012" href="#0081-SUCM-012">0081-SUCM-012</a>)
```

> 

Reviewer 2: ```suggestion

Propose a successor market which specifies a parent which is settled, and for which the successor time window has expired. The proposal is declined. (<a name="0081-SUCM-018" href="#0081-SUCM-018">0081-SUCM-018</a>)
```

> When fetching a market that is part of a "family tree", we should see both the parent and each successor `marketID` (<a name="0081-SUCM-013" href="#0081-SUCM-013">0081-SUCM-013</a>)

Reviewer 2: ```suggestion
When fetching a market that is part of a "parent / successor chain", we should see both the parent and each successor `marketID` (<a name="0081-SUCM-013" href="#0081-SUCM-013">0081-SUCM-013</a>)
```

### Snapshots / checkpoints

> After a checkpoint restart the child parent market state is preserved where applicable inc. the LPs ELS	(<a name="0081-SUCM-016" href="#0081-SUCM-016">0081-SUCM-016</a>)

Reviewer 2: ```suggestion
After a LNL checkpoint restart the successor (child) / parent market state is preserved where applicable inc. the LPs ELS	(<a name="0081-SUCM-016" href="#0081-SUCM-016">0081-SUCM-016</a>)
```

> After snapshot restart the child parent market state is preserved where applicable inc. the LPs ELS	(<a name="0081-SUCM-017" href="#0081-SUCM-017">0081-SUCM-017</a>)

Reviewer 2: ```suggestion
After snapshot restart the successor (child) / parent market state is preserved where applicable inc. the LPs ELS	(<a name="0081-SUCM-017" href="#0081-SUCM-017">0081-SUCM-017</a>)
```

- Both `A` and `B` submit a liquidity commitment of `s1` and `s2` to the new market before the opening auction ends. No other LP submits liquidity to the new market. Then, once the opening auction resolved the LPs `A` and `B` have virtual stakes `v1` and `v2` (<a name="0081-SUCM-020" href="#0081-SUCM-020">0081-SUCM-020</a>).
- As above but `A` submits `s1` and `B` doesn't submit anything. Then `A` has virtual stake `v1` and `B` has virtual stake `0` (<a name="0081-SUCM-021" href="#0081-SUCM-021">0081-SUCM-021</a>).
- As above but `A` submits more than `s1`. Then `A` has virtual stake larger than `v1`. (<a name="0081-SUCM-022" href="#0081-SUCM-022">0081-SUCM-022</a>)

### Equity like share

> With parent and child markets running at the same time. Ensure that the child ELS matches the parent at point of child leaving opening auction - trade on both markets and Ensure that markets diverge over subsequent trading (<a name="0081-SUCM-015" href="#0081-SUCM-015">0081-SUCM-015</a>)

Reviewer 2: This duplicates 020 above.
